File: snippets/signature.py
import hmac
from base64 import urlsafe_b64encode
from hashlib import sha256

API_KEY = b'23456789'
API_SECRET = b'k69x50j0'
EXPIRE_AT = b'1893456000'

digest = hmac.new(API_SECRET, API_KEY + EXPIRE_AT, sha256).digest()
signature = urlsafe_b64encode(digest).rstrip(b'=')

print('Signature: {}'.format(signature.decode()))

File: snippets/signature.java
import java.util.Base64;

import javax.crypto.Mac;
import javax.crypto.spec.SecretKeySpec;

import org.apache.commons.lang3.StringUtils;

class SignatureSnippet {
    public static void main(String args[]) throws Exception {

        String API_KEY = "23456789";
        String API_SECRET = "k69x50j0";
        String EXPIRE_AT = "1893456000";

        String algorithm = "HmacSHA256";
        byte[] key = API_SECRET.getBytes();
        byte[] msg = (API_KEY + EXPIRE_AT).getBytes();

        Mac mac = Mac.getInstance(algorithm);
        SecretKeySpec secretKeySpec = new SecretKeySpec(key, algorithm);
        mac.init(secretKeySpec);
        byte[] digest = mac.doFinal(msg);

        byte[] b64Bytes = Base64.getEncoder().encode(digest);
        String b64Str = new String(b64Bytes);
        String signature = StringUtils.stripEnd(b64Str.replace('+', '-').replace('/', '_'), "=");

        System.out.format("Signature: %s\n", signature);
    }
}
